jwst/extract_1d/extract.py

```diff
@@ -133,31 +133,41 @@ def open_extract1d_ref(refname: str, exptype: str) -> dict:
         handle for the jwst.datamodels object for the extract1d file.
     """
 
+    # the extract1d reference file can be 1 of three types:  'json', 'fits', or  'asdf'
+    refname_type = refname[-4:].lower()
     if refname == "N/A":
         ref_dict = None
     else:
-        fd = open(refname)
-        try:
-            ref_dict = json.load(fd)
-            ref_dict['ref_file_type'] = FILE_TYPE_JSON
-            fd.close()
-        except (UnicodeDecodeError, JSONDecodeError):
-            # input file is not JSON so lets try fits and then asdf
-            fd.close()
+        if refname_type == 'json':
+            fd = open(refname)
             try:
-                fd = fits.open(refname)
-                fits_present = 1
+                ref_dict = json.load(fd)
+                ref_dict['ref_file_type'] = FILE_TYPE_JSON
                 fd.close()
-            except OSError:
-                fits_present = 0
-            if fits_present:
+            except (UnicodeDecodeError, JSONDecodeError):
+                # Input file does not load correctly as json file.
+                # Probably an error in json file
+                fd.close()
+                log.error(f"Extract 1d json reference file has an error, run a json validator off line and fix the file")
+                raise RuntimeError("Invalid json extract 1d reference file, run json validator off line and fix file.")
+        elif refname_type == 'fits':
+            try:
+                fd = fits.open(refname)
                 extract_model = datamodels.MultiExtract1dImageModel(refname)
                 ref_dict = {'ref_file_type': FILE_TYPE_IMAGE, 'ref_model': extract_model}
-            else:
-                extract_model = datamodels.Extract1dIFUModel(refname)
-                ref_dict = dict()
-                ref_dict['ref_file_type'] = FILE_TYPE_ASDF
-                ref_dict['ref_model'] = extract_model
+                fd.close()
+            except OSError:
+                log.error(f"Extract 1d fits reference file has an error")
```

Comment:
Same here: "extract1d FITS reference file ..."   (FITS should always be caps, because it's an acronym)